app.js

```diff
@@ -0,0 +1,138 @@
+    if (!time.match(military) || frequency < 0 || !Number.isInteger(frequency)) {
+        document.getElementById("message").innerHTML = "Invalid Input!"
+        console.log("Error");
+        return false;
+    } else {
+        userRef.push({
+          name: document.getElementById("trainName").value.trim(),
+          destination: document.getElementById("destination").value.trim(),
+          time: time,
+          frequency: frequency,
+        })
+        document.getElementById("trainForm").reset()
+    }
+    retreiveData()
+  }
+//Retreive Data
+function retreiveData() {
+    userRef.on('value', function(data){
+        document.querySelector("#trainList").innerHTML = ""
+        const trains = data.val()
+    for (const key in trains) {
+        if (trains.hasOwnProperty(key)) {
+            const train = trains[key];
+            let x = moment(train.time, "HH:mm")
+            let y = moment()
+            let difference = y.diff(x, "minutes")
+            let minutesAway = train.frequency - (difference%train.frequency)
+            let nextTrain = moment().add(minutesAway, 'minutes').format("hh:mm A");
+            
+            if (difference < 0) {
+                nextTrain = moment(train.time, "HH:mm").format("hh:mm A")
+                minutesAway = Math.abs(difference)
+            } 
+            
+            let trainElement = document.createElement('tr')
+                trainElement.innerHTML = `
+                    <td class="editable">${train.name}</td>
+                    <td class="editable">${train.destination}</td>
+                    <td class="editable">${train.frequency}</td>
+                    <td class="editable">${nextTrain}</td>
+                    <td>${minutesAway}</td>
+                    <td><div class="delete"><a href="#"><i class="material-icons">delete</i></a></div></td>
+                `
+            document.querySelector('#trainList').appendChild(trainElement)
+        }  
+    }
+    })
+}
+
+
+// Edit Train Schedule
+function edit(editIcon) {
+    let editArr = document.querySelectorAll('.editable')
+    editArr.forEach(function(el){
+        el.contentEditable = "true"
+        el.style.color = "brown"
+    })
+    editIcon.innerHTML =`
+    <a href="#"><i class="material-icons done">done</i></a>
+    `
+    let deleteArr = document.querySelectorAll('.delete')
+    deleteArr.forEach(function(el){
+        el.style.visibility = "visible"
+    })
+}
+
+//Update Data
+function updateData(doneIcon) {
+    let editArr = document.querySelectorAll('.editable')
+    editArr.forEach(function(el){
+        el.contentEditable = "false"
+        el.style.color = "black"
+    })
+    doneIcon.innerHTML =`
+    <a href="#"><i class="material-icons edit">edit</i></a>
+    `
+    let deleteArr = document.querySelectorAll('.delete')
+    deleteArr.forEach(function(el){
+        el.style.visibility = "hidden"
+    })
+}
+
+//EventListener
+document.addEventListener("DOMContentLoaded", function(){
```

Comment:
I'm not sure what the point of this whole section is. It looks like it's polling for data, I think listeners is the more appropriate way to handle this.